exercises/simple-linked-list/example.scala

```diff
@@ -0,0 +1,94 @@
+import scala.reflect.ClassTag
+
+class Node[+T]
+case object Empty extends Node[Nothing]
+case class NonEmpty[T](value: T, var next: Node[T]) extends Node[T]
+
+trait SimpleLinkedList[T] {
+  def isEmpty: Boolean
+  def value: T
+  def add(item: T): SimpleLinkedList[T]
+  def next: SimpleLinkedList[T]
+  def reverse: SimpleLinkedList[T]
+  def toArray[U >: T](implicit m: ClassTag[U]): Array[U]
```

Comment:
toArray[T] does not make the compiler happy...
Here is some background -
* http://stackoverflow.com/questions/6085085/why-cant-i-create-an-array-of-generic-type
* https://www.chrisstucchio.com/blog/2014/cannot_find_classtag_for_element_type_T.html

I can simplify the function though..